velox/functions/prestosql/Bitwise.h

```diff
@@ -28,6 +30,31 @@ struct BitwiseAndFunction {
   }
 };
 
+template <typename T>
+struct BitCountFunction {
+  template <typename TInput>
+  FOLLY_ALWAYS_INLINE bool call(int64_t& result, TInput num, TInput bits) {
+    if (bits == kMaxBits) {
+      result = std::bitset<kMaxBits>(num).count();
+      return true;
+    }
+   if (bits <= 1 || bits > kMaxBits) {
+      result = 0;
+      return false;
+    }
+    // check if input "num" falls within the limits of max and min that
+    // can be represented with "bits".
+    int64_t lowbitsmask = (1L << bits) - 1;
```

Comment:
As per https://github.com/prestodb/presto/blob/41329cdf883f264ede6e53c7f1d66e1297c5c0b3/presto-main/src/main/java/com/facebook/presto/operator/scalar/BitwiseFunctions.java#L39, the lowbitsmask should be int64_t lowbitsmask = (1L << (bits - 1)) - 1;

Any reason that was changed ?

Reply:
I was testing out the difference. I will undo it. The logic expects the MSB to be unused. Hence (bits-1).